kythe/cxx/indexer/cxx/IndexerASTHooks.cc

```diff
@@ -4251,10 +4251,10 @@ IndexerASTVisitor::BuildNodeIdForTemplateName(const clang::TemplateName& Name) {
             // templates.
             if (const auto* FTD = dyn_cast<clang::FunctionTemplateDecl>(
                     Name.getAsTemplateDecl())) {
-              for (const auto* ID = FTD;
-                   (ID = ID->getInstantiatedFromMemberTemplate()); FTD = ID)
+              while ((FTD = FTD->getInstantiatedFromMemberTemplate())) {
```

Comment:
I think you can drop one set of parens here

Reply:
This results in a compiler warning about using the result of an assignment in a condition (which is intentional here).

Reply:
fair enough :)